habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -61,9 +64,21 @@ def if_config_to_lower(config):
         else:
             return config
 
+    has_ignores = ignore_keys is None
     for attr, value in config_from.items():
-        if hasattr(config_to, attr.lower()):
-            setattr(config_to, attr.lower(), if_config_to_lower(value))
+        low_attr = attr.lower()
+        if has_ignores or low_attr not in ignore_keys:
+            if hasattr(config_to, low_attr):
+                setattr(config_to, low_attr, if_config_to_lower(value))
+            else:
+                # TODO consider new type of Error for this?
+                raise RuntimeError(
+                    f"""{low_attr} is not found on habitat_sim but is found on habitat_lab config.
+                    It's also not in the list of keys to ignore: {ignore_keys}
+                    Did you make a typo in the YAML config?
```

Comment:
Source of the config can be different.
```suggestion
                    Did you make a typo in the config?
```